plugins/osdn/proxy.go

```diff
@@ -24,11 +28,17 @@ type ovsProxyPlugin struct {
 	baseEndpointsHandler pconfig.EndpointsConfigHandler
 }
 
-func CreateProxyPlugin(registry *Registry) (api.FilteringEndpointsConfigHandler, error) {
-	return &ovsProxyPlugin{
-		registry: registry,
-		podsByIP: make(map[string]*kapi.Pod),
-	}, nil
+// Called by higher layers to create the proxy plugin instance; only used by nodes
+func NewProxyPlugin(pluginType string, osClient *osclient.Client, kClient *kclient.Client) (api.FilteringEndpointsConfigHandler, error) {
+	switch strings.ToLower(pluginType) {
+	case MultiTenantPluginName:
+		return &ovsProxyPlugin{
+			registry: NewRegistry(osClient, kClient),
+			podsByIP: make(map[string]*kapi.Pod),
+		}, nil
+	}
```

Comment:
If we expose isMultitenantPlugin() in plugins/osdn/plugin.go, then we can use it here:
```
if IsMultitenantPlugin(pluginType) {
   return &ovsProxyPlugin{...}
}
```